client/my-sites/checkout/checkout/checkout-container.jsx

```diff
@@ -12,24 +14,47 @@ import Checkout from '../checkout';
 import CartData from 'components/data/cart';
 import CheckoutData from 'components/data/checkout';
 import SecondaryCart from '../cart/secondary-cart';
+import SignupSiteCreatedNotice from 'my-sites/checkout/checkout/signup-site-created-notice';
 
 /**
  * Style dependencies
  */
 import './checkout-container.scss';
 
+/**
+ * Returns whether given site has been created in the last `n` minutes
+ *
+ * @param  {String}  createdAt               The site creation date stamp
+ * @param  {Number}  creationWindowInMinutes A site is considered 'new' if it's been created in this time window (in minutes)
+ * @return {Boolean}                         If the site is 'new'. Default `false`
+ */
+function isSelectedSiteNew( createdAt, creationWindowInMinutes = 5 ) {
+	return moment( createdAt ).isAfter( moment().subtract( creationWindowInMinutes, 'minutes' ) );
+}
+
 class CheckoutContainer extends React.Component {
-	state = {
-		headerText: '',
-	};
+	constructor( props ) {
+		super( props );
+		this.state = {
```

Comment:
We should still be able to achieve the same initial state assignment without dropping in to the `constructor` pattern here - was there some trouble with adding the `shouldDisplaySiteCreatedNotice` in the `state` property initializer? In case it's the prop usage, we can still access initial props in property initializers like so:

```js
state = {
  headerText: '',
  shouldDisplaySiteCreatedNotice:
    this.props.isComingFromSignup && isSelectedSiteNew( get( this.props, 'selectedSite.options.created_at', '' ) )
};
```

Not a blocker, just an FYI :)

Reply:
Good call. I think it's a bad habit of mine. I'll test this out and rework it. 

Thanks!!!